tee_supplicant/src/tee_supplicant.c

```diff
@@ -324,6 +325,51 @@ static void tee_supp_main(void *p1, void *p2, void *p3)
 	}
 }
 
+static char *_strdup(const char *str)
+{
+	int len;
+	char *s;
+
+	len = strlen(str)+1;
+	s = k_malloc(len);
+	strcpy(s, str);
+	return s;
+}
+
+static int do_mkdir(const char *path)
+{
+	int rc;
+
+	rc = fs_mkdir(path);
+	if (rc == 0 || rc == -EEXIST) {
+		rc = 0;
+	}
+
+	return rc;
+}
+
+static int mkpath(const char *path)
+{
+	int status = 0;
+	char *subpath = _strdup(path);
+	char *prev = subpath;
+	char *curr = NULL;
+
+	while (status == 0 && (curr = strchr(prev, '/')) != 0) {
```

Comment:
Again, there can be weird cases with paths like `////` or `/../../`. Now, when base TEE storage path is user-configurable, there will be someone who'll that option to `/tee//`.

Reply:
Yes. I have tested.